promql/engine.go

```diff
@@ -668,16 +691,19 @@ func (enh *EvalNodeHelper) signatureFunc(on bool, names ...string) func(labels.L
 
 // rangeEval evaluates the given expressions, and then for each step calls
 // the given function with the values computed for each expression at that
-// step.  The return value is the combination into time series of  of all the
+// step.  The return value is the combination into time series of all the
 // function call results.
 func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ...Expr) Matrix {
 	numSteps := int((ev.endTimestamp-ev.startTimestamp)/ev.interval) + 1
 	matrixes := make([]Matrix, len(exprs))
 	origMatrixes := make([]Matrix, len(exprs))
+	tempNumSamples := ev.currentSamples
 	for i, e := range exprs {
 		// Functions will take string arguments from the expressions, not the values.
 		if e != nil && e.Type() != ValueTypeString {
+			tempNumSamples = ev.currentSamples
```

Comment:
This wipes out the tempNumSamples if there's more than one expr, you don't need this line

Reply:
I don't understand, each call to eval is adding samples to `ev.currentSamples`, but after the call we want the result of `ev.currentSamples` to be the value prior to the `eval` call plus the amount of samples returned.

Reply:
as we discussed here: https://github.com/prometheus/prometheus/pull/4513#discussion_r218979274

Reply:
After the call, the value of ev.currentSamples should already be the eval call plus the amount of samples returned - as that's the contract.